awscli/customizations/s3/filegenerator.py

```diff
@@ -151,8 +151,8 @@ def call(self, files):
 
     def _inject_extra_information(self, file_stat_kwargs, extra_information):
         src_type = file_stat_kwargs['src_type']
-        file_stat_kwargs['size'] = extra_information['Size']
-        file_stat_kwargs['last_update'] = extra_information['LastModified']
+        file_stat_kwargs['size'] = extra_information.get('Size')
```

Comment:
Assuming that consuming code of `FileStat` can gracefully handle when these values are `None`, then this looks good to me.